test/remote/gateways/remote_flex_charge_test.rb

```diff
@@ -147,6 +147,14 @@ def test_failed_fetch_access_token
     assert_match(/400/, error.message)
   end
 
+  def test_successful_purchase_with_token
+    store = @gateway.store(@credit_card_cit, {})
+    assert_success store
+
```

Comment:
You could include assertions representing the store transaction (like a typical response field). (If it's something possible)

Reply:
The only thing in the response that allows us to identify is a store. It's the token and that would be used to set the authorization parameter of the response. So, if this field is not present, blank, or wrong the purchase will fail.